src/gamekit.cs

```diff
@@ -1709,10 +1716,19 @@ interface GKAchievement : NSSecureCoding {
 		[Export ("initWithIdentifier:player:")]
 		NativeHandle Constructor ([NullAllowed] string identifier, GKPlayer player);
 
-		[NoWatch]
+#if XAMCORE_5_0
 		[MacCatalyst (13, 1)]
-		[Async (ResultTypeName = "GKChallengeComposeResult")]
+		[Deprecated(PlatformName.iOS, 17, 0)]
+		[Deprecated(PlatformName.MacOSX, 14, 0)]
+		[Deprecated(PlatformName.TvOS, 17, 0)]
+		[Deprecated(PlatformName.MacCatalyst, 17, 0)]
 		[Export ("challengeComposeControllerWithMessage:players:completionHandler:")]
+#else
+		[TV (17, 0), iOS (17, 0), Mac (14, 0), MacCatalyst (17, 0)]
+		[Export ("challengeComposeControllerWithMessage:players:completion:")]
+#endif
+		[NoWatch]
+		[Async (ResultTypeName = "GKChallengeComposeResult")]
```

Comment:
This also needs to be bound as two different methods, same way it was done in GKLeaderboardEntry below.